tests/components/zwave/test_init.py

```diff
@@ -1310,699 +1235,848 @@ def mock_connect(receiver, signal, *args, **kwargs):
+    assert len(node.refresh_value.mock_calls) == 2
+    assert (
+        sorted(
+            [
+                node.refresh_value.mock_calls[0][1][0],
+                node.refresh_value.mock_calls[1][1][0],
+            ]
+        )
+        == sorted([value.value_id, power_value.value_id])
+    )
+
+
+async def test_refresh_node(hass, mock_openzwave, zwave_setup):
+    """Test zwave refresh_node service."""
+    zwave_network = hass.data[DATA_NETWORK]
+    zwave_network.state = MockNetwork.STATE_READY
+
+    await hass.async_start()
+
+    node = MockNode(node_id=14)
+    zwave_network.nodes = {14: node}
+    await hass.services.async_call("zwave", "refresh_node", {const.ATTR_NODE_ID: 14})
+    await hass.async_block_till_done()
+
+    assert node.refresh_info.called
+    assert len(node.refresh_info.mock_calls) == 1
+
+
+async def test_set_node_value(hass, mock_openzwave, zwave_setup):
+    """Test zwave set_node_value service."""
+    zwave_network = hass.data[DATA_NETWORK]
+    zwave_network.state = MockNetwork.STATE_READY
+
+    await hass.async_start()
+
+    value = MockValue(index=12, command_class=const.COMMAND_CLASS_INDICATOR, data=4)
+    node = MockNode(node_id=14, command_classes=[const.COMMAND_CLASS_INDICATOR])
+    node.values = {12: value}
+    node.get_values.return_value = node.values
+    zwave_network.nodes = {14: node}
+
+    await hass.services.async_call(
+        "zwave",
+        "set_node_value",
+        {
+            const.ATTR_NODE_ID: 14,
+            const.ATTR_VALUE_ID: 12,
+            const.ATTR_CONFIG_VALUE: 2,
+        },
+    )
+    await hass.async_block_till_done()
+
+    assert zwave_network.nodes[14].values[12].data == 2
+
+
+async def test_set_node_value_with_long_id_and_text_value(
+    hass, mock_openzwave, zwave_setup
+):
+    """Test zwave set_node_value service."""
+    zwave_network = hass.data[DATA_NETWORK]
+    zwave_network.state = MockNetwork.STATE_READY
+
+    await hass.async_start()
+
+    value = MockValue(
+        index=87512398541236578,
+        command_class=const.COMMAND_CLASS_SWITCH_COLOR,
+        data="#ff0000",
+    )
+    node = MockNode(node_id=14, command_classes=[const.COMMAND_CLASS_SWITCH_COLOR])
+    node.values = {87512398541236578: value}
+    node.get_values.return_value = node.values
+    zwave_network.nodes = {14: node}
+
+    await hass.services.async_call(
+        "zwave",
+        "set_node_value",
+        {
+            const.ATTR_NODE_ID: 14,
+            const.ATTR_VALUE_ID: "87512398541236578",
+            const.ATTR_CONFIG_VALUE: "#00ff00",
+        },
+    )
+    await hass.async_block_till_done()
+
+    assert zwave_network.nodes[14].values[87512398541236578].data == "#00ff00"
+
+
+async def test_refresh_node_value(hass, mock_openzwave, zwave_setup):
+    """Test zwave refresh_node_value service."""
+    zwave_network = hass.data[DATA_NETWORK]
+    zwave_network.state = MockNetwork.STATE_READY
+
+    await hass.async_start()
```

Comment:
I see this in every test. Should this also be added to `zwave_setup` ?

Reply:
It's in about 50% of the tests. It seems a bit overdone to call `hass.async_start()` in about 20 tests when not needed. We could make 2 fixtures, one with, one without?

Reply:
I like that.

Reply:
Done